lib/process-services/task-list/components/task-filters.component.ts

```diff
@@ -151,22 +131,47 @@ export class TaskFiltersComponent implements OnInit, OnChanges {
         this.appsProcessService.getDeployedApplicationsByName(appName).subscribe(
             application => {
                 this.getFiltersByAppId(application.id);
-                this.selectTaskFilter(this.filterParam, this.filters);
             },
             (err) => {
                 this.error.emit(err);
             });
     }
 
+    /**
+     * Create default filters by appId
+     * @param appId
+     */
+    getDefaultFilters(appId?: number) {
+        this.taskFilterService.createDefaultFilters(appId).subscribe(
+            (resDefault: FilterRepresentationModel[]) => {
+                this.resetFilter();
+                resDefault.forEach((filter) => {
+                    this.filters.push(filter);
+                });
+
+                this.selectTaskFilter(this.filterParam, this.filters);
+                this.success.emit(resDefault);
+            },
+            (errDefault: any) => {
+                this.error.emit(errDefault);
+            }
+        );
+    }
+
     /**
      * Pass the selected filter as next
      * @param filter
      */
     public selectFilter(filter: FilterRepresentationModel) {
         this.currentFilter = filter;
         this.filterClick.emit(filter);
+        this.router.navigate(['/activiti/apps', this.appId || 0, 'tasks', filter.id]);
```

Comment:
this is wrong the component should never know what a navigation is. The parent should take care of it.

Reply:
Done.